### PR TITLE
FC-0001: remove legacy waffle flags

### PR DESCRIPTION
**Pull request by dyudyunov**
This is a part of https://github.com/openedx/public-engineering/issues/28

Remove all the LegacyWaffle* classes

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dyudyunov! I've created [OSPR-6637](https://openedx.atlassian.net/browse/OSPR-6637) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@dyudyunov Thank you for the contribution. Is this ready for our review?

**Comment by dyudyunov** [comment]
@natabene yes, this one is ready along with https://github.com/openedx/edx-ora2/pull/1853 and https://github.com/openedx/edx-platform/pull/30330

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,10 @@ Change Log
 Unreleased
 ~~~~~~~~~~
 
+[5.0.0] - 2022-04-22
+--------------------
+
+* Removed LegacyWaffle* classes
```
Comment (robrap) [inline CHANGELOG.rst:20]: Reminder to search the whole repository. There is some how-to doc clean-up needed.

Comment (dyudyunov) [reply CHANGELOG.rst:20]: Yeah, thanks
Removed the LegacyWaffle* section from the how-to doc

Comment (robrap) [reply CHANGELOG.rst:20]: Do you mind updating as follows?
```suggestion
* BREAKING CHANGE: Removed LegacyWaffle* classes. Although this is a breaking change, all known uses have already been fixed.
```

**Review by robrap — COMMENTED** [review]
I think this is almost ready, but we can merge/release it after all uses have been removed. Feel free to squash when you are ready.

File: CHANGELOG.rst (hunk 1 above)
Comment (dyudyunov) [reply CHANGELOG.rst]: Done!)

**Comment by dyudyunov** [comment]
@robrap added fixup and squashed, now it's ready to be merged! 🚀

**Review by robrap — APPROVED** [review]
Looks great. Thanks. I am going to adjust the commit comment a little when I merge.

**Comment by openedx-webhooks** [comment]
@dyudyunov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-toggles&entry.867055334=https://github.com/openedx/edx-toggles/pull/211&entry.1484655318=https://github.com/dyudyunov&entry.752974735=2022-04-25+07:45&entry.1917517419=2022-05-11+19:02&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**